File: backend/app/schemas/export.py
from pydantic import BaseModel
from datetime import datetime


class ExportRequest(BaseModel):
    translation_id: int
    format: str  # pdf, word, img


class ExportResponse(BaseModel):
    id: int
    translation_id: int
    format: str
    file_path: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/export.ts
import axios from "axios";

const api = axios.create({
  baseURL: "/api/v1",
  headers: { "Content-Type": "application/json" },
});

export interface ExportRequest {
  translation_id: number;
  format: "pdf" | "word" | "img";
}

export interface ExportResponse {
  id: number;
  translation_id: number;
  format: string;
  file_path: string;
  created_at: string;
}

export async function createExport(req: ExportRequest): Promise<ExportResponse> {
  const { data } = await api.post<ExportResponse>("/exports/", req);
  return data;
}

export function getExportDownloadUrl(exportId: number): string {
  return `/api/v1/exports/download/${exportId}`;
}
